auth/builtin.go

```diff
@@ -59,13 +59,28 @@ func (r *NoAuth) Refresh(refresh string) (token Token, err error) {
 }
 
 // Builtin auth provider.
-type Builtin struct {
+type Builtin struct{}
+
+// Parse Token out of a string
+func ParseToken(requestToken string) (string, error) {
+	splitToken := strings.Split(requestToken, " ")
+	if splitToken[0] != "Bearer" {
+		return "", errors.New("authentication header not of type bearer")
+	}
+	token := strings.TrimSpace(splitToken[1])
+	if len(token) < 1 {
+		return "", errors.New("no authentication header found")
+	}
+
+	return token, nil
 }
 
 // Authenticate the token
 func (r *Builtin) Authenticate(request *Request) (jwToken *jwt.Token, err error) {
-	token := strings.Replace(request.Token, "Bearer", "", 1)
-	token = strings.Fields(token)[0]
+	token, err := ParseToken(request.Token)
+	if err != nil {
+		return nil, err
```

Comment:
When named return values (pattern is used) the explicit assigned return values here is redundant.  `err` is already assigned and jwtToken is already nil.

Reply:
Since parseToken() returns an error, the call should happen after the defer so that it is logged.